beprof/curve.py

```diff
@@ -113,7 +113,13 @@ def change_domain(self, domain):
             raise ValueError('in change_domain():' 'the old domain does not include the new one')
 
         y = np.interp(domain, self.x, self.y)
-        obj = self.__class__(np.stack((domain, y), axis=1), **self.__dict__['metadata'])
+        # We need to join together domain and y because we are recreating 2 dimensional Curve object
+        # (we pass it as argument to self.__class__ to do so and it takes 2 or 3 dimensional arrays as argument.
```

Comment:
We should have a discussion on curve with 3-D data. Check if it is implemented correctly, test methods etc... - separate issue.